### PR TITLE
fix(client): throw error if constructor not initialized properly

### LINKED ISSUE #2319: Throw if missing constructor requirements
Throw / warn if missing items from the constructor that are required (i.e. links / cache)

### PR DESCRIPTION
**Pull request by peggyrayzis**
- throw an error if client is not initialized w/ link or cache
- added `graphql` as a dev dependency to `apollo-client`
- added postinstall script for lerna bootstrap

Closes #2319

### CODE REVIEW

File: packages/apollo-client/src/ApolloClient.ts [hunk 1]
```diff
@@ -91,6 +91,15 @@ export default class ApolloClient<TCacheShape> implements DataProxy {
       defaultOptions,
     } = options;
 
+    if (!link || !cache) {
+      throw new Error(`
+        In order to initialize Apollo Client, you must specify link & cache properties on the config object.
+        For more information, please visit:
+          http://dev.apollodata.com/react/initialization.html#creating-client
```
Comment (jbaxleyiii) [inline packages/apollo-client/src/ApolloClient.ts]: Let's change the link to what will be the new one (https://apollographql.com/docs/react/setup)

**Comment by peggyrayzis** [comment]
@jbaxleyiii good idea, just changed it! 🌮 :shipit:

**Comment by jbaxleyiii** [comment]
@peggyrayzis awesome! If you could just add to the changelog I'll merge this!

**Comment by peggyrayzis** [comment]
Added! Post-Summit, we should discuss automating the changelog 😀 With all these packages, it will prob make our lives easier haha.

https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-cli

**Pull request merged**